app/controllers/webauthn_verifications_controller.rb

```diff
@@ -25,8 +33,7 @@ def authenticate
     @verification.generate_otp
     @verification.expire_path_token
 
-    # TODO: render html with webauthn verification otp instead of json
-    render json: { message: "success" }
+    redirect_to(URI.parse("http://localhost:#{port}\?code=#{@verification.otp}").to_s, allow_other_host: true)
   rescue WebAuthn::Error => e
     render json: { message: e.message }, status: :unauthorized
```

Comment:
Can be done in a follow up: Can we change these to return plain text? That way the return format will always be consistent among the endpoint.

Same goes for https://github.com/Shopify/rubygems.org/blob/accept-port/app/controllers/webauthn_verifications_controller.rb#L61-L62
Maybe we can switch on format type (html - we redirect, plain text - return a message).

The format type would need to be modified in the routes too: https://github.com/Shopify/rubygems.org/blob/0ea7c5e02c46498a39aeedc3f34101ac95fee49d/config/routes.rb#L190-L191
(The TODO says to change it to HTML, but that was when we were intending to do the manual paste OTP solution and not sockets)

Reply:
👍🏾 I've updated the TODOs here so we can get back to this in a follow up